src/main.c

```diff
@@ -123,23 +123,7 @@ int_t main(int argc, char *argv[])
         return -1;
     }
 
-    /* platform specific init */

```

Comment:
I would like to keep this idea. I may add an array with a list of fixed directories to check. This way each build could be used with a relative path or absolute path.

Reply:
It still seems a bit odd that there is a runtime platform switch, when there are different builds for different platforms anyway. With how it is right now, there is no way really to change that directory at compile time. Even if I override BASE_PATH it'll get overridden here. So at least for non windows the build option has no effect at all.

Reply:
BASE_PATH is empty by default. If it is empty, and only then, it will fallback to this logic.
If you override BASE_PATH this block will be completely ignored.

Reply:
You're right. I didn't think it through. The issue I observed was the other way around: Having those relative directories didn't work anymore. This also keeps the "config" part, so the files would be in /usr/local/etc/teddycloud/conf, while I'd remove the conf part.
Also from our last discussion I thought the BASE_PATH is basically for the other data (which would go to /usr/local/share/teddycloud).

Reply:
I like the idea with the multiple search paths though, but I think changing the base_path is looking at it from the wrong direction. Unusally we should check some predefined dirs for the config file. Based on where we found it, we can come up with  a reasonable default for the base_path, but the user may change that anyway if they liked to by changing stuff in the ini file. Maybe we should look at it from a requirements view. For me there are not many:
- The configuration files shall be located in /usr/local/etc/teddycloud (not in another subfolder). I'm happy to either inject that at compile time, have it as a default already, or have a runtime flag.
- The "other" directories shall be expected in /usr/local/share/teddycloud (in subfolders like data, certs, etc) per default. Same deal as with the configuration files.

I can update this PR with something that's making that possible, but at the moment I don't see how I would attack this with the current solution (apart from patching it pre build, but that'd be my least preferred option)

I'm sure you have a lot more things that are important to you with this, since I'm new to the party and have no idea about other users needs (apart from maybe "don't break my current setup). Would you be able to write down your expectations? I would try to come up with another proposal based on that then.

Reply:
You're right. I should be able to get my stuff running without deleting this part.

Reply:
It makes sense to split data / certs and config. 

## teddyCloud should:
- be able to run with a relative path (data/certs/config folders beside the executable), as "portable app"
- be able to run with fixed path in /usr/local/* (unix style)
- be able to run with fixed path in /teddycloud/* (docker)
- detect if the fixed path is available, otherwise fall back to relative variant.

In the future I may add a cli option to set the --config-dir and the --data-cert-dir


PS: Thank you for taking the time =).